language-server/src/jsonc-instance.js

```diff
@@ -196,6 +196,41 @@ export class JsoncInstance {
   textLength() {
     return this.node.length;
   }
+
+  keyAtPosition(position) {
+    const positionOffset = this.textDocument.offsetAt(position);
+
+    if (this.typeOf() !== "object") {
+      return undefined;
+    }
+
+    for (const propertyNode of this.node.children) {
+      const propertyNameNode = propertyNode.children[0];
+      const propertyStartOffset = propertyNameNode.offset;
+      const propertyEndOffset = propertyStartOffset + propertyNameNode.length;
+
+      if (positionOffset >= propertyStartOffset && positionOffset <= propertyEndOffset) {
+        let pointer;
+        if (this.pointer === "") {
+          pointer = `/${propertyNameNode.value}`;
+        } else {
+          pointer = `${this.pointer}/${propertyNameNode.value}`;
+        }
```

Comment:
There are some complexities to building JSON Pointers. This won't always work. Use the `append` function from `@hyperjump/json-pointer` for this.